ownership/views.py

```diff
@@ -1,194 +1,279 @@
+        if not self.error_details:
+            self.handle_post()
+        return self._respond()
+
+
+class PromptView(DeliveryViewBase):
     """
-    Lookup url for name and append transfer_id to url.
+    Initial landing view, prompting for accept or decline button
+    Posts to ProcessDeliveryView
     """
-    url = reverse(name)
-    if transfer_id:
-        url = "{}?transfer_id={}".format(url, transfer_id)
-    return url
+    http_method_names = ['get']
+    template_name = 'ownership/index.html'
 
 
-def response_with_delivery(request, param_dict, func):
+class ProcessView(DeliveryViewBase):
     """
-    Pull out transfer_id from request params and return func(delivery).
-    If delivery already complete render already complete message.
-    Renders missing authorization transfer_id message otherwise.
+    Handles POST from PromptView.
+    If user clicked decline, redirects to the decline form.
+    If clicked accept, process acceptance and redirect to accepted page
     """
-    transfer_id = param_dict.get('transfer_id', None)
-    if transfer_id:
+    http_method_names = ['post']
+
+    def process_accept(self):
+        delivery = self.delivery
+        request = self.request
+        if delivery.is_complete():
+            self.set_already_complete_error()
+            return
         try:
-            details = DeliveryDetails.from_transfer_id(transfer_id, request.user)
-            delivery = details.get_delivery()
-            # update the status
-            if delivery.is_complete():
-                return render_already_complete(request, delivery)
-            return func(request, delivery)
-        except ObjectDoesNotExist:
-            return general_error(request, msg=TRANSFER_ID_NOT_FOUND, status=404)
-        except DataServiceError as err:
-            return general_error(request, msg=(err), status=500)
-    else:
-        return general_error(request, msg=MISSING_TRANSFER_ID_MSG, status=400)
-
-
-def render_already_complete(request, delivery):
+            delivery_util = self.delivery_type.make_delivery_util(delivery, request.user)
+            delivery_util.accept_project_transfer()
+            delivery_util.share_with_additional_users()
+            warning_message = delivery_util.get_warning_message()
+            message = self.delivery_type.make_processed_message(delivery, request.user, 'accepted', warning_message=warning_message)
+            self.warning_message = warning_message
+            message.send()
+            delivery.mark_accepted(request.user.get_username(), message.email_text)
+        except DataServiceError as e:
+            self.set_error_details(500, 'Unable to transfer ownership: {}'.format(e.message))
+        except Exception as e:
+            self.set_error_details(500, str(e))
+
+    def handle_post(self):
+        request = self.request
+        if 'decline' in request.POST:
+            # Redirect to decline page
+            self.set_redirect('ownership-decline')
```

Comment:
I think this type of logic is also needed for the GET.

Reply:
Can you elaborate why? This view (ProcessView) is meant to be post-only. I'll add logic to ensure that's the case but am curious why it should support GET.